p2p/net/network.go

```diff
@@ -154,18 +155,16 @@ func (n *Net) IncomingMessages() []chan IncomingMessageEvent {
 }
 
 // SubscribeClosingConnections registers a channel where closing connections events are reported
-func (n *Net) SubscribeClosingConnections() chan Connection {
+func (n *Net) SubscribeClosingConnections(f func(connection Connection)) {
 	n.clsMutex.Lock()
-	ch := make(chan Connection, 20) // todo: set a var for the buf size
-	n.closingConnections = append(n.closingConnections, ch)
+	n.closingConnections = append(n.closingConnections, f)
 	n.clsMutex.Unlock()
-	return ch
 }
 
 func (n *Net) publishClosingConnection(connection Connection) {
 	n.clsMutex.RLock()
-	for _, c := range n.closingConnections {
-		c <- connection
+	for _, f := range n.closingConnections {
+		f(connection)
```

Comment:
f can block? is this ok if it blocks? if not- maybe run f in go routine

Reply:
it should block.